src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -113,6 +122,9 @@ CATCH_RETURN()
     RETURN_HR_IF_NULL(E_INVALIDARG, columns);
     *columns = 0;
 
+    _formatInUse = _format.Get();
+    _fontInUse = _font.Get();

```

Comment:
I'm just using the "normal" font variant in the `GetColumns` method since it has no knowledge of whether the text is going to be italic or not. I believe it's just used to determine whether a glyph is wide or not, which shouldn't differ between font variants. To be honest, I would have expected this to be entirely independent of the font, otherwise I don't see how it can work in a terminal environment, but that's somebody else's problem.

Reply:
This is one of the big historic gotchas that we inherited. The font in use controls the width of "ambiguous" glyphs _only in the traditional console_, and in the case of raster fonts it also changes the return values from certain readback APIs. It's a madhouse, but this is one that's required for compatibility.

Conpty intentionally breaks that and forces all ambiguous glyphs to be narrow, because a sane terminal emulator will do the same.

Reply:
(in Terminal, the font will never be queried for glyph width:)

https://github.com/microsoft/terminal/blob/b8e6b8e27c2c902ce18798904ce09ff32d2f68fd/src/cascadia/TerminalControl/TermControl.cpp#L47-L62

Reply:
Ah OK. That makes sense. Thanks for the explanation.